### PR TITLE
Add suppport for TestFx to record with user account

### PR DESCRIPTION
**Pull request by vidai-msft**
## Description

Support recording scenarios test cases via user account.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [ ] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [x] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: documentation/testing-docs/using-azure-test-framework.md [hunk 1]
```diff
@@ -127,27 +135,26 @@ AZURE_TEST_MODE=Playback
...
+- [Run the tests](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/azure-powershell-developer-guide.md#recordingrunning-tests) and make sure that you have a generated `.json` file that corresponds to the test name, and it should be located under the `SessionRecords` folder within the test project.
 - If you want to switch from Record to Playback or from Playback to Record, consider below steps.
-  - If you opt for JSON config file, update the value of the property `HttpRecorderMode` in the JSON.
-  - If you opt for environment variables
-    - If you have `AZURE_TEST_MODE` set, update the value of this variable
-    - Otherwise, update the value of the property `HttpRecorderMode` defined in the variable `TEST_CSM_ORGID_AUTHENTICATION`
+  - If you choose to use a JSON config file, update the value of the property `HttpRecorderMode` in the JSON file.
+  - If you prefer environment variables
+    - If you have `HttpRecorderMode` defined in the environment variable `TEST_CSM_ORGID_AUTHENTICATION`, update its value accordingly.
+    - Otherwise, update the value of `AZURE_TEST_MODE`.
```
Comment (NoriZC) [inline documentation/testing-docs/using-azure-test-framework.md:157]: Do we optimize these 2 options? I see that you reverse their order, but it seems the second one here is simpler.

File: documentation/testing-docs/using-azure-test-framework.md [hunk 2]
```diff
@@ -1,30 +1,31 @@
 # Using Azure PowerShell Test Framework
 
-- [Using Microsoft.Rest.ClientRuntime.Azure.TestFramework](#using-microsoftrestclientruntimeazuretestframework)
+- [Using Azure PowerShell Test Framework](#using-azure-powershell-test-framework)
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply documentation/testing-docs/using-azure-test-framework.md:3]: Microsoft.Rest.ClientRuntime.Azure is obsolete. The TestFramework does not have any dependencies on that library. So we can no longer use that namespace.

File: documentation/testing-docs/using-azure-test-framework.md [hunk 3]
```diff
@@ -1,30 +1,31 @@
...
-      - [Record Test with service principal](#record-test-with-service-principal)
-      - [Playback Test](#playback-test)
+      - [Record Tests](#record-tests)
+      - [Playback Tests](#playback-tests)
   - [JSON Config File V.S. Environment Variables](#json-config-file-vs-environment-variables)
   - [Record or Playback Tests](#record-or-playback-tests)
   - [Change Test Environment settings at run-time](#change-test-environment-settings-at-run-time)
-    - [Once you set your connection string, you can add or update key/value settings](#once-you-set-your-connection-string-you-can-add-or-update-keyvalue-settings)
-      - [Note:](#note)
+      - [Once you set your connection string, you can add or update key/value settings](#once-you-set-your-connection-string-you-can-add-or-update-keyvalue-settings)
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply documentation/testing-docs/using-azure-test-framework.md:18]: This is expected. Not sure if this is caused by markdown or VS code formatting.

File: documentation/testing-docs/using-azure-test-framework.md (hunk 1 above)
Comment (vidai-msft) [reply documentation/testing-docs/using-azure-test-framework.md:157]: `AZURE_TEST_MODE` is a redundant option. I cannot remove it due to compatibility issues. But `HttpRecorderMode` in either `testcredentials.json` or environment variable `TEST_CSM_ORGID_AUTHENTICATION` has higher priority than `AZURE_TEST_MODE`.

File: tools/Modules/TestFx-Tasks.psm1 [hunk 4]
```diff
@@ -17,7 +17,7 @@ $script:TestFxEnvExtraPropKeys = @(
 )
 
 function Set-TestFxEnvironment {
-    [CmdletBinding(DefaultParameterSetName = "NewServicePrincipal")]
+    [CmdletBinding(DefaultParameterSetName = "UserAccount")]
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply tools/Modules/TestFx-Tasks.psm1:20]: This function only needs to run once and should not be part of the automation scripts. The breaking change won't have any impact to partner teams.

File: documentation/testing-docs/using-azure-test-framework.md [hunk 5]
```diff
@@ -1,30 +1,31 @@
...
+- [Using Azure PowerShell Test Framework](#using-azure-powershell-test-framework)
   - [Getting Started](#getting-started)
   - [Azure PowerShell Test Framework](#azure-powershell-test-framework)
   - [Setup prior to Record or Playback of tests](#setup-prior-to-record-or-playback-of-tests)
-    - [Run Command Set-TestFxEnvironment to Build Connection String (Recommended)](#run-command-set-testfxenvironment-to-build-connection-string-recommended)
-      - [Create New Service Principal](#create-new-service-principal)
+    - [Run Command Set-TestFxEnvironment to Build Connection String](#run-command-set-testfxenvironment-to-build-connection-string)
+      - [Use user account to record test cases (Recommended)](#use-user-account-to-record-test-cases-recommended)
+      - [Create New Service Principal (Not Recommended)](#create-new-service-principal-not-recommended)
       - [Use Existing Service Principal](#use-existing-service-principal)
```
[the remark this answers is not in the document]
Comment (vidai-msft) [reply documentation/testing-docs/using-azure-test-framework.md]: Mark it as not recommended as well. Thanks!

**Review by NoriZC — APPROVED** [review]

**Pull request merged**